### PR TITLE
[SC 16234] Two small fixes to built-in ValidMind tests

### PR DESCRIPTION
**Pull request by AnilSorathiya**
# Pull Request Description

## What and why?

**IQROutliersBarPlot** — Boolean and binary columns (≤2 unique values) are now excluded from plots and from the raw outlier summary table. Before, they could appear in raw data or cause quantile errors on boolean dtypes. After, only meaningful numeric features are analyzed.

```python
import numpy as np
import pandas as pd
import validmind as vm
df = pd.DataFrame({
    "numeric": np.random.randn(100),
    "flag": np.random.choice([True, False], 100),
})
dataset = vm.init_dataset(input_id="ds", dataset=df, __log=False)
result = vm.tests.run_test(
    "validmind.data_validation.IQROutliersBarPlot",
    inputs={"dataset": dataset},
)
result.show()
```

**WeakspotsDiagnosis** — Custom `thresholds` can now specify only some metrics (e.g. `{"accuracy": 0.65}`). Before, missing keys could break plots or pass/fail logic. After, plots use defaults for metrics without a custom threshold (reference lines still show), and pass/fail only checks the thresholds you provide.
```python
result = vm.tests.run_test(
    "validmind.model_validation.sklearn.WeakspotsDiagnosis",
    inputs={"datasets": [train_ds, test_ds], "model": model},
    params={"thresholds": {"accuracy": 0.65}},  # only accuracy used for pass/fail
)
```
## How to test

```bash
pytest tests/unit_tests/data_validation/test_IQROutliersBarPlot.py -v
pytest tests/unit_tests/model_validation/sklearn/test_WeakspotsDiagnosis.py -v
```

Manual checks (optional):

- Run `IQROutliersBarPlot` on a dataset with boolean columns; confirm they are not in the bar plots or raw outlier table.
- Run `WeakspotsDiagnosis` with `thresholds={"accuracy": 0.65}`; confirm plots show reference lines for all metrics and pass/fail only uses accuracy.

## What needs special review?

- `WeakspotsDiagnosis`: split between `plot_thresholds` (defaults + overrides) and `pass_thresholds` (user-only when custom thresholds are passed).
- `IQROutliersBarPlot`: `eligible_columns` filter applied consistently to plots and raw data.

## Release notes

- **IQROutliersBarPlot:** Boolean and binary features are excluded from outlier analysis and summary output.
- **WeakspotsDiagnosis:** You can pass partial custom thresholds; plots still show default reference lines for other metrics, and pass/fail only evaluates the thresholds you set.

## Checklist

- [x] What and why
- [ ] Screenshots or videos (Frontend) — N/A
- [x] How to test
- [x] What needs special review
- [x] Dependencies, breaking changes, and deployment notes
- [x] Labels applied
- [x] PR linked to Shortcut (SC-16234)
- [x] Unit tests added (Backend)
- [ ] Tested locally
- [ ] Documentation updated (if required)
- [ ] Environment variable additions/changes documented (if required)

### CODE REVIEW

**Review by johnwalz97 — APPROVED** [review]
nice!

**Review by juanmleng — APPROVED** [review]
Fantastic, thanks @AnilSorathiya!
